### DISCUSSION TITLE
Disable analytics when working local

### DISCUSSION THREAD

**Discussion by Matthijz98** [squidfunk/mkdocs-material#6441] [Ideas]
I am probably missing something but how would i disable parts of my mkdocs.yml when working localy?
I know of the group plugin but that only works in the plugins part of the config file?

I could override the analytics js to add a simple check but feels a bid hacky.

**Comment by alexvoss** [comment]
One option if you cannot solve this with environment variables is [configuration inheritance](https://www.mkdocs.org/user-guide/configuration/#configuration-inheritance), which would mean having a local `mkdocs.yml` and one for 'non-local', with both inheriting from a common one.

Not entirely sure if there perhaps a better one. Perhaps someone else will chime in.

**Comment by squidfunk** [comment]
The approach we've been following for the last few years is what @alexvoss suggested – use an environment variable and define it as a secret in your CI build. This ensures that analytics are only collected for the live site.

https://github.com/squidfunk/mkdocs-material/blob/d363f903b863df0990b49627804debc4e730e589/mkdocs.yml#L104-L106

For GitHub Actions, set `GOOGLE_ANALYTICS_KEY` to your property.

**Reply by Matthijz98** [reply]
That is a good option.
Thanks
